### PR TITLE
Provide more details for invalid value argument on Set function.

### PR DESCRIPTION
**Pull request by justinpark0916**
If the value argument is not of the same type of the variable, user only gets a message saying that the argument type is invalid, without indication of the expected type.

With this change, when `Set()` function has invalid types on its arguments, the error message will display more details on which type is problematic.

Addressing this issue: https://github.com/microsoft/Power-Fx/issues/736

### CODE REVIEW

**Comment by justinpark0916** [comment]
> @justinpark0916 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="{Microsoft}"]

**Comment by justinpark0916** [comment]
> @microsoft-github-policy-service agree company="Microsoft"

@microsoft-github-policy-service agree company="Microsoft"

**Comment by justinpark0916** [comment]
> @justinpark0916 the command you issued was incorrect. Please try again.
> 
> Examples are:
> 
> ```
> @microsoft-github-policy-service agree
> ```
> 
> and
> 
> ```
> @microsoft-github-policy-service agree company="your company"
> ```

@microsoft-github-policy-service agree company="Microsoft"

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs [hunk 1]
```diff
@@ -76,7 +76,7 @@ public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[
 
                     if (!arg0.Accepts(arg1))
                     {
-                        errors.EnsureError(DocumentErrorSeverity.Critical, args[1], ErrBadType);
+                        errors.EnsureError(DocumentErrorSeverity.Critical, args[1], ErrBadType_ExpectedType_ProvidedType, arg0.GetKindString(), arg1.GetKindString());
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs:79]: >args[1] [](http://example.com/codeflow?start=75&length=7)

Let's change this to args1 (variable).

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/SetFunctionTests.cs [hunk 2]
```diff
@@ -385,6 +385,22 @@ public static ReadOnlySymbolTable NewMutableFromRecord(RecordType type)
         {
             var symTable = ReadOnlySymbolTable.NewFromRecord(type, allowMutable: true);
             return symTable;
+        }
+
+        [Fact]
+        public void NewErrorMessageWithTypes()
```
Comment (anderson-joyle) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/SetFunctionTests.cs:391]: >NewErrorMessageWithTypes [](http://example.com/codeflow?start=20&length=24)

Add another test case exploring a different locale. For that you can provide a CultureInfo to PowerFxConfig object.
`var powerFxConfig = new PowerFxConfig(CultureInfo.CreateSpecificCulture("fr-FR"));`

**Review by anderson-joyle — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**